networkit/cpp/embedding/AliasSampler.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * AliasSampler.cpp
+ *
+ *  Created on: 03.07.2020
+ *      Author: Klaus Ahrens  <ahrens@informatik.hu-berlin.de>
+ *
+ *  adapted and reimplemented in C++17
+ *  from node2vec [https://arxiv.org/pdf/1607.00653v1.pdf]
+ *
+ */
+
+// networkit-format
+
+#include <mutex>
+#include <random>
+#include <vector>
+
+#include <networkit/Globals.hpp>
+#include <networkit/auxiliary/Random.hpp>
+
+#include "AliasSampler.hpp"
+
+namespace NetworKit {
+namespace Embedding {
+
+std::mt19937_64 &getURNG() {
+    static bool seedPerThread = true;
+    static std::once_flag once;
+    std::call_once(once, [] { Aux::Random::setSeed(1, seedPerThread); });
```

Comment:
It is quite unexpected for a user that calling the embedding algorithm changes the global seed. If for some reason you require a custom seed, please do not use the global URNG of NetworKit but use a custom one.

Reply:
Agreed, library functions should not change the global seed.